packages/lib/src/vis/shared/TooltipMesh.tsx

```diff
@@ -49,11 +53,19 @@ function TooltipMesh(props: Props) {
     [worldToData, showTooltip]
   );
 
-  // Hide tooltip when pointer leaves mesh or user starts panning
-  const onPointerOut = useCallback(hideTooltip, [hideTooltip]);
-  const onPointerDown = useCallback(hideTooltip, [hideTooltip]);
+  // Hide tooltip when pointer leaves mesh
+  const onPointerOut = useCallback(() => {
+    /* `onPointerOut` is called after `onPointerUp` for some reason,
+     * so we make sure not to hide the tooltip again in this case. */
+    if (tooltipOpen) {
+      hideTooltip();
+    }
+  }, [hideTooltip, tooltipOpen]);
+
+  // Hide tooltip when user starts panning
+  const onPointerDown = useCallback(() => hideTooltip(), [hideTooltip]);
 
-  // Show tooltip after dragging unless pointer has left canvas
+  // Show tooltip after dragging, if pointer is released inside the vis viewport
```

Comment:
`canvas` is the wrong term here. On a heatmap with a ratio, the tooltip remains hidden if the pointer is released outside of the vis (i.e. outside of the heatmap mesh). The term `vis` would also be incorrect as we want the tooltip to remain hidden when the vis is zoomed in and the pointer is released outside of the canvas (whereas the vis technically extends past the canvas). `vis viewport` seems more accurate - i.e. the area of the vis located inside the canvas (and therefore visible to the user).

Reply:
or `vis visible area`. `viewport` is fine though